test/jdk/com/sun/management/HotSpotDiagnosticMXBean/DumpThreadsWithEliminatedLock.java

```diff
@@ -0,0 +1,171 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8356870
+ * @summary Test HotSpotDiagnosticMXBean.dumpThreads with a thread owning a monitor for
+ *     an object that is scalar replaced
+ * @requires !vm.debug & (vm.compMode != "Xcomp")
+ * @requires (vm.opt.TieredStopAtLevel == null | vm.opt.TieredStopAtLevel == 4)
+ * @modules jdk.management
+ * @library /test/lib
+ * @run main/othervm DumpThreadsWithEliminatedLock plain platform
+ * @run main/othervm DumpThreadsWithEliminatedLock plain virtual
+ * @run main/othervm DumpThreadsWithEliminatedLock json platform
+ * @run main/othervm DumpThreadsWithEliminatedLock json virtual
+ */
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.lang.management.ManagementFactory;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.List;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Stream;
+import com.sun.management.HotSpotDiagnosticMXBean;
+import jdk.test.lib.threaddump.ThreadDump;
+import jdk.test.lib.thread.VThreadRunner;
+
+public class DumpThreadsWithEliminatedLock {
+
+    public static void main(String[] args) throws Exception {
+        boolean plain = switch (args[0]) {
+            case "plain" -> true;
+            case "json"  -> false;
+            default      -> throw new RuntimeException("Unknown dump format");
+        };
+
+        ThreadFactory factory = switch (args[1]) {
+            case "platform" -> Thread.ofPlatform().factory();
+            case "virtual"  -> Thread.ofVirtual().factory();
+            default         -> throw new RuntimeException("Unknown thread kind");
+        };
+
+        // need at least two carriers for JTREG_TEST_THREAD_FACTORY=Virtual
+        if (Thread.currentThread().isVirtual()) {
+            VThreadRunner.ensureParallelism(2);
+        }
+
+        // A thread that spins creating and adding to a StringBuffer. StringBuffer is
+        // synchronized, assume object will be scalar replaced and the lock eliminated.
+        var done = new AtomicBoolean();
+        var ref = new AtomicReference<String>();
+        Thread thread = factory.newThread(() -> {
+            while (!done.get()) {
+                StringBuffer sb = new StringBuffer();
+                sb.append(System.currentTimeMillis());
+                String s = sb.toString();
+                ref.set(s);
```

Comment:
```suggestion
                ref.set(
                        new StringBuffer()
                                .append(System.currentTimeMillis())
                                .toString());
```

Reply:
This looks a lot less readable so I don't think we should change it.